pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -572,12 +578,15 @@ protected boolean createNewSegmentMetadataZNRecord(TableConfig realtimeTableConf
     final String newZnodePath =
         ZKMetadataProvider.constructPropertyStorePathForSegment(realtimeTableName, newSegmentNameStr);
 
-    if (!isConnected() || !isLeader()) {
-      // We can potentially log a different value than what we saw ....
-      LOGGER.warn("Lost leadership while committing new segment metadata for {} for table {}: isLeader={}, isConnected={}",
-          newSegmentNameStr, rawTableName, isLeader(), isConnected());
-      _controllerMetrics.addMeteredGlobalValue(ControllerMeter.CONTROLLER_NOT_LEADER, 1L);
-      return false;
+    if (!isNewTableSetup) {
```

Comment:
I am not able to follow the need for this new boolean. It looks like this method is called from only one place. If thats the case, why don't we remove the isLeader check completely

Reply:
Saw the other place where its called from but I still don't get why the check is not done by the caller before invoking this method.
 Are we using isNewTableSetup as a proxy for isLeader() check ?

Reply:
The intention is to do it just before we write the metadata to the property store, which happens to be the last step of the method